lib/insights/api/common/filter.rb

```diff
@@ -14,21 +14,25 @@ class Filter
         #   An AR model that acts as the base collection to be filtered
         # raw_filter::
         #   The filter from the request query string
-        # api_doc_definition::
-        #   The documented object definition from the OpenAPI doc
+        # api_doc_definitions::
+        #   The documented object definitions from the OpenAPI doc
         # extra_filterable_attributes::
         #   Attributes that can be used for filtering but are not documented in the OpenAPI doc.  Something like `{"undocumented_column" => {"type" => "string"}}`
         #
         # == Returns:
         # A new Filter object, call #apply to get the filtered set of results.
         #
-        def initialize(model, raw_filter, api_doc_definition, extra_filterable_attributes = {})
-          self.query                   = model
-          @api_doc_definition          = api_doc_definition
+        def initialize(model, raw_filter, api_doc_definitions, extra_filterable_attributes = {})
+          @raw_filter = raw_filter
+          self.query = if filter_associations.present?
```

Comment:
Having a method call inside `initialize` is not great.  What do you think of splitting it into a method and removing the attr_reader?

```ruby
def query
  @query ||= filter_associations.present? ? model.left_outer_joins(filter_associations) : model
end
```